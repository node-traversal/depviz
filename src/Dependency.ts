export default class Dependency {
    private readonly _name: string;
    private readonly _deps: Dependency[];

    constructor(name: string, deps: Dependency[]) {
        this._name = name;
        this._deps = deps;
    }

    public get name(): string {
        return this._name;
    }

    public get deps(): Dependency[] {
        return this._deps;
    }
}