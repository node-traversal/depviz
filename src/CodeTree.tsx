import * as React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import TreeView from '@material-ui/lab/TreeView';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import ChevronRightIcon from '@material-ui/icons/ChevronRight';
import CodeTreeNodes from './CodeTreeNodes';
import Dependency from "./Dependency";

const useStyles = makeStyles({
    root: {
        height: 216,
        flexGrow: 1,
        maxWidth: 400,
        'text-align': 'initial'
    },
});

interface CodeTreeProps {
    dependencies: Dependency[];
}

const CodeTree: React.FC<CodeTreeProps> = ({dependencies}) => {
    const classes = useStyles();

    return (
        <TreeView
            className={classes.root}
            // defaultExpanded={['a', '5', '6']}
            defaultCollapseIcon={<ExpandMoreIcon />}
            defaultExpandIcon={<ChevronRightIcon />}>
            <CodeTreeNodes nodes={dependencies} path="" />
        </TreeView>
    );
};

export default CodeTree;