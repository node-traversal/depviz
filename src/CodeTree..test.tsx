import React from 'react';
import CodeTree from './CodeTree';
import { shallow } from 'enzyme';
import Dependency from "./Dependency";

it('renders code tree', () => {
    const dependencies = [
        new Dependency('the-node-name', [])
    ];

    const wrapper = shallow(<CodeTree dependencies={dependencies}/>);
    expect(wrapper.debug()).toMatchSnapshot();
});
