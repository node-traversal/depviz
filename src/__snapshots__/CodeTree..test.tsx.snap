// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders code tree 1`] = `
"<WithStyles(ForwardRef(TreeView)) className=\\"makeStyles-root-1\\" defaultCollapseIcon={{...}} defaultExpandIcon={{...}}>
  <CodeTreeNodes nodes={{...}} path=\\"\\" />
</WithStyles(ForwardRef(TreeView))>"
`;
